src/input.c

```diff
@@ -427,7 +427,7 @@ void _glfwInputGamepad(_GLFWjoystick *js)
         GLFWgamepadstate state;
         if (glfwGetGamepadState(jid, &state))
         {
-            _glfw.callbacks.gamepad_state(jid, state.buttons, state.axes);
+            _glfw.callbacks.gamepad_state(jid, state);
```

Comment:
since the callback references the state, thought it would be better to pass the state itself, not the buttons and axes (easier mapping to the struct in other libraries like lwjgl)